### PR TITLE
Sub-pel refinement optimization

### PR DESCRIPTION
**Pull request by NaderMahdi**
**Sub-pel refinement optimization**
Restructure the Sub-pel refinement to be done in an NxM window around the best integer-pel and where NxM is set to 3x3 by default.

The recorded data on the 360p clips is 11% speedup in encoding time for 0.1% bdrate gain and no impact on the memory.

### CODE REVIEW

File: Source/Lib/Common/Codec/EbMotionEstimation.c [hunk 1]
```diff
@@ -4198,7 +4198,521 @@ static void FullPelSearch_LCU(MeContext *context_ptr, uint32_t listIndex,
...
+                    *p_best_mv =
+                        ((uint16_t)half_mv_y[3] << 16) | ((uint16_t)half_mv_x[3]);
+                }
+            }
+#endif
+            // TL position
+            search_region_index =
+                search_index_x +
+                (int16_t)context_ptr->interpolated_stride * search_index_y;
+            distortion_topleft_position =
```
Comment (kylophone) [inline Source/Lib/Common/Codec/EbMotionEstimation.c]: Avoid all these ternarys. Could be written in a more readable fashion using if/else blocks or maybe a switch.

**Comment by mafonsoN** [comment]
@NaderMahdi can you please clarify the changes here compared to #330 ?

File: Source/Lib/Common/Codec/EbMotionEstimation.c (hunk 1 above)
Comment (NaderMahdi) [reply Source/Lib/Common/Codec/EbMotionEstimation.c]: Addressed - Thanks.

**Comment by NaderMahdi** [comment]
> @NaderMahdi can you please clarify the changes here compared to #330 ?
@mafonsoN basically no difference, just the implementation is different. #330 is more a simulation code hoewever the #464 is production code.

**Pull request closed without merging**